File: Games/EchoGame/src/main/java/jokatu/game/games/echo/game/Echo.java
package jokatu.game.games.echo.game;

import jokatu.game.event.PublicGameEvent;
import jokatu.game.games.echo.input.EchoInput;
import jokatu.game.player.StandardPlayer;
import org.jetbrains.annotations.NotNull;

public class Echo implements PublicGameEvent {

	private final String message;

	public Echo(EchoInput input, StandardPlayer player) {
		message = player.getName() + " said: " + input.getString();
	}

	@NotNull
	@Override
	public String getMessage() {
		return message;
	}
}

File: Games/EchoGame/src/main/kotlin/jokatu/game/games/echo/game/Echo.kt
package jokatu.game.games.echo.game

import jokatu.game.event.PublicGameEvent
import jokatu.game.games.echo.input.EchoInput
import jokatu.game.player.StandardPlayer

/**
 * An event that echoes the users input to everyone in the game.
 */
class Echo(input: EchoInput, player: StandardPlayer) : PublicGameEvent {

	override val message: String = player.name + " said: " + input.string

}
